src/js/views/globals/root_views.js

```diff
@@ -113,6 +114,12 @@ const ModalRegionView = TopRegionView.extend({
   },
 });
 
+const ModalLoadingRegionView = ModalRegionView.extend({
```

Comment:
I think in this case we should go ahead and remove `ModalSidebarRegionView` and make both `contains` and `setLocation` options that can be passed in on instantiation.

Also I think the `contains` should be the same function that's on the preloader region.  It's a little unorthodox, but I think we could do something like:
```js
function preventRegionClose() {
  return this.getRegion('region').hasView();
}
```
Then `contains: preventRegionClose,` for the preload region and the `modalLoading`

Reply:
Is this what you had in mind for this: https://github.com/RoundingWell/care-ops-frontend/compare/d2a60b3358e3ffbf471d940bcda967b7ebc4afa8..24c037a16d0ddd932548942b1d6fd3b72919da15?